### PR TITLE
🐛 Use leveled scoring for Code Review check

### LINKED ISSUE #2302: Feature: Code Review Check handle commits made by version bump bots
**Is your feature request related to a problem? Please describe.**

Discussion started at https://github.com/electron/electron/pull/35741#pullrequestreview-1118960274

Just oppening the discussion about the commits made automatically by some bots that, depending on its access rights and on its complexity, can not represent a big security risk.

We have some situations where it is not reasonable for the project to review every single commit made by the bot, since it is a very frequent and automatic job. But, by the other hand, allowing an ignore list for bots also leaves the door open to other projects excluding bots that might be more susceptible to compromise.

**Describe the solution you'd like**
The Code Review Check could handle, somehow, commits made by version bump bots, that just updates the version of the project itself, not of its dependencies.

**Describe alternatives you've considered**
I've consider to allow the user to exclude commits made by the bot, but this has the turnaround mentioned above of leaving the door open to other projects excluding bots that might be more susceptible to compromise.

**Additional context**

### LINKED ISSUE #2450: Wrong Scorecards result in CodeReview check due to Dependabot/RenovateBot PR reviews
**Describe the bug**
A single maintainer project will return incorrect result on CodeReview check if that single maintainer is reviewing Dependabot/RenovateBot PRs. This was indicated by Sonatype folks who ran Scorecards on a large number of maven projects.

**Reproduction steps**
Try a single maintainer repo that uses Dependabot/RenovateBot

**Expected behavior**
Ignore Dependabot and RenovateBot PRs completely from processing in CodeReview check. Don't count them as code reviews. Codereview are usually intended to check human reviews on human created PRs.

**Additional context**

### PR DESCRIPTION
**Pull request by raghavkaul**
#### What kind of change does this PR introduce?
Update the Code-Review check to better account for bot activity
Also updates Raw Results.

#### What is the new behavior (if this is a feature change)?**
* Scoring changes: Leveled scoring
  * If any changes haven't been reviewed, reduce score by 7 points
    * If multiple changes haven't been reviewed, give a minimum score
  * If any bot changes that haven't been reviewed, reduce score by 3 points
* If the only activity found on a repo is bot commit activity, scoring is inconclusive.

- [x] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes
Fixes #2450.
Fixes #2302.

#### Does this PR introduce a user-facing change?
```release-note
Code-Review check uses leveled scoring
Code-Review check returns inconclusive results when recent activity is solely made up of bot commits
```

### CODE REVIEW

File: clients/githubrepo/graphql.go [hunk 1]
```diff
@@ -396,12 +397,14 @@ func commitsFrom(data *graphqlData, repoOwner, repoName string) ([]clients.Commi
 				string(pr.Repository.Name) != repoName {
 				continue
 			}
+			openedByBot := strings.HasPrefix(string(pr.Author.ResourcePath), "/apps/")
```
Comment (laurentsimon) [inline clients/githubrepo/graphql.go:400]: Can you leave a comment about what it looks like for common bots like, dependabot, renovate?

File: checks/evaluation/code_review_test.go [hunk 2]
```diff
@@ -55,10 +55,47 @@ func TestCodeReview(t *testing.T) {
...
 					},
 					{
-						Commits: []clients.Commit{{SHA: "1"}},
+						Commits: []clients.Commit{{SHA: "a"}},
+					},
+				},
+			},
+		},
+		{
+			name: "all human changesets reviewed, missing review on bot changeset",
```
Comment (laurentsimon) [inline checks/evaluation/code_review_test.go:67]: shall we add a few more tests to exercise all the paths in the switch statement that computes the score?

File: pkg/json_raw_results.go [hunk 3]
```diff
@@ -549,33 +549,20 @@ func (r *jsonScorecardRawResult) setDefaultCommitData(changesets []checker.Chang
...
-						Login: r.Author.Login,
-					},
-				})
-			}
+		for j := range cs.Reviews {
+			r := cs.Reviews[j]
+			reviews = append(reviews, jsonReview{
+				State: r.State,
+				Reviewer: jsonUser{
+					Login: r.Author.Login,
```
Comment (laurentsimon) [inline pkg/json_raw_results.go:557]: shall we add the isBot as an additional field?

File: docs/checks.md [hunk 4]
```diff
@@ -207,13 +207,13 @@ malicious code (either as a malicious contributor or as an attacker who has
 subverted a contributor's account), because a reviewer might detect the
 subversion.
 
-The check first tries to detect whether [Branch-Protection](checks.md#branch-protection) is enabled on the
-default branch with at least one required reviewer. If this fails, the check
-determines whether the most recent (~30) commits have a Github-approved review
+The check determines whether the most recent (~30) commits have a Github-approved review
```
Comment (laurentsimon) [inline docs/checks.md:210]: Let's add a note about how the score is computed

**Review by laurentsimon — APPROVED** [review]
Thanks. LGTM. Added a few comments.

File: clients/githubrepo/graphql.go (hunk 1 above)
Comment (raghavkaul) [reply clients/githubrepo/graphql.go:402]: It's usually just `/apps/<app name>`. Left a comment.

**Review by laurentsimon — APPROVED** [review]

File: e2e/code_review_test.go [hunk 5]
```diff
@@ -104,5 +104,47 @@ var _ = Describe("E2E TEST:"+checks.CheckCodeReview, func() {
...
+			Expect(scut.ValidateTestReturn(nil, "use code reviews", &expected, &result, &dl)).Should(BeTrue())
+			Expect(repoClient.Close()).Should(BeNil())
+		})
+		It("Should return minimum score for a single-maintainer project with some unreviewed human changesets", func() {
+			dl := scut.TestDetailLogger{}
+			repo, err := githubrepo.MakeGithubRepo("Kromey/fast_poisson")
+			Expect(err).Should(BeNil())
+			repoClient := githubrepo.CreateGithubRepoClient(context.Background(), logger)
+			err = repoClient.InitRepo(repo, clients.HeadSHA, 0)
+			Expect(err).Should(BeNil())
```
Comment (spencerschrock) [inline e2e/code_review_test.go:134]: We don't control this repo, and this test runs off `HEAD`. We shouldn't rely on it in an E2E test if the results can change day to day. Theoretically the repo could also be deleted one day, which would also cause a failure.

For now can you add a commit SHA that exhibits the behavior you want, like the test above?

**Review by spencerschrock — CHANGES_REQUESTED** [review]
Overall looks good.

File: e2e/code_review_test.go (hunk 5 above)
Comment (raghavkaul) [reply e2e/code_review_test.go]: Pinned this test on a SHA

**Review by laurentsimon — APPROVED** [review]

**Comment by raghavkaul** [comment]
Some unit test failures, looking

**Review by spencerschrock — APPROVED** [review]
Thanks for pinning the E2E test

**Review by laurentsimon — APPROVED** [review]

**Review by spencerschrock — APPROVED** [review]

**Pull request merged**